src/wallet/silentpayments.h

```diff
@@ -0,0 +1,93 @@
+
+using PubKey = std::variant<CPubKey, XOnlyPubKey>;
+
+/**
+ * @brief Generate a silent payment labeled address.
+ *
+ * @param receiver The receiver's silent payment destination (i.e. scan and spend public keys).
+ * @param label The label tweak
+ * @return V0SilentPaymentDestination The silent payment destination, with `B_spend -> B_spend + label * G`.
+ *
+ * @see ComputeSilentPaymentLabelTweak(const CKey& scan_key, const int m);
+ */
+V0SilentPaymentDestination GenerateSilentPaymentLabeledAddress(const V0SilentPaymentDestination& receiver, const uint256& label);
+
+/**
+ * @brief Generate silent payment taproot destinations.
+ *
+ * Given a set of silent payment destinations, generate the requested number of outputs. If a silent payment
+ * destination is repeated, this indicates multiple outputs are requested for the same recipient. The silent payment
+ * desintaions are passed in map where the key indicates their desired position in the final tx.vout array.
+ *
+ * @param priv_tweak_data                     The private tweak data. This is a pair containing a private key and input hash.
+ *                                            The private key is either the receiver's scan key or the sum of the sender's
+ *                                            input private keys (excluding inputs not eligible for shared secret derivation).
+ *                                            The input hash is the tagged hash of hash of the smallest outpoint and sum
+ *                                            of the eligible input public keys.
+ * @param sp_dests                            The silent payment destinations.
+ * @return std::map<size_t, WitnessV1Taproot> The generated silent payment taproot destinations.
+ *
+ */
+std::map<size_t, WitnessV1Taproot> GenerateSilentPaymentTaprootDestinations(const BIP352::PrivTweakData& priv_tweak_data, const std::map<size_t, V0SilentPaymentDestination>& sp_dests);
+
+/**
+ * @brief Get silent payment tweak data from transaction inputs.
+ *
+ * Get the necessary data from the transaction inputs to be able to scan the transaction outputs for silent payment outputs.
+ * This requires knowledge of the prevout scriptPubKey, which is passed via `coins`.
+ *
+ * If the transaction has silent payment eligible inputs, the public keys from these inputs are returned as a sum along with
+ * the transaction input hash. If there are no eligible inputs, nullopt is returned, indicating that this transaction does not
+ * contain silent payment outputs.
+ *
+ * @param vin                                              The transaction inputs.
+ * @param coins                                            The coins (potentially) spent in this transaction.
+ * @return std::optional<std::pair<uint256, PubTweakData>> The silent payment tweak data, or nullopt if not found.
+ */
+std::optional<BIP352::PubTweakData> GetSilentPaymentTweakDataFromTxInputs(const std::vector<CTxIn>& vin, const std::map<COutPoint, Coin>& coins);
```

Comment:
Assuming I still need this for the index #28241, I still prefer if this wasn't part of the wallet. Afaik nothing in this file has a dependency on the wallet, so the easiest change would to be move this to `src/common`.

Reply:
Now that we have `src/bip352.h`, it might make more sense to get rid of `src/wallet/silentpayents.h` and put everything in `src/bip352.h`. In the future, anything that is wallet specific can go in `src/wallet/silentpayments.h` and you'd have everything you need for an index in `src/bip352.h`. Thoughts?